crates/data_components/src/oracle/connection.rs

```diff
@@ -101,22 +110,39 @@ impl OracleConnectionParamsBuilder {
 
     #[must_use]
     pub fn build(self) -> OracleConnectionParams {
+        let connect_string = format!(
+            "//{}:{}/{}",
+            self.port.unwrap_or(1521),
+            self.host,
+            self.service_name.unwrap_or_else(|| "XEPDB1".to_string())
+        );
+
         OracleConnectionParams {
-            host: self.host,
             username: self.username,
             password: self.password,
-            port: self.port.unwrap_or(1521),
-            service_name: self.service_name.unwrap_or_else(|| "XEPDB1".to_string()),
+            connect_string,
         }
     }
 }
 
-pub async fn connect(params: &OracleConnectionParams) -> super::Result<OracleConnectionPool> {
-    let connect_string = format!("//{}:{}/{}", params.host, params.port, params.service_name);
+pub async fn connect(
+    params: &OracleConnectionParams,
+    tns_admin: Option<&str>,
+) -> super::Result<OracleConnectionPool> {
+    if let Some(tns_admin) = tns_admin {
+        // Initializes Oracle client library with the specified TNS_ADMIN directory
+        // Note: this is applied for the first connection only, if library is already initialized, dynamically changing TNS_ADMIN has no effect
+        let _ = oracle::InitParams::new()
+            .oracle_client_config_dir(tns_admin)
+            .context(OracleInitSnafu)?
+            .init()
+            .context(OracleInitSnafu)?;
```

Comment:
```suggestion
        let already_initialized = oracle::InitParams::new()
            .oracle_client_config_dir(tns_admin)
            .context(OracleInitSnafu)?
            .init()
            .context(OracleInitSnafu)?;
      if already_initialized {
          tracing::warn!("Oracle connection already initialized. Updating 'tns_admin' will have no effect.");
      }
```

Maybe something like this if it makes sense.

Reply:
Updated, thank you! Warning might confuse when multiple datasets are initialized (same wallet config). Added the following tracing 

```rust
if initialized_here {
            tracing::info!("Using wallet directory for Oracle data connector: {wallet_path}");
        } else {
            tracing::debug!(
                "Oracle client library was already initialized, using existing configuration"
            );
        }
```